fp_app/app/models/appointment.rb

```diff
@@ -8,11 +8,14 @@ class Appointment < ApplicationRecord
   validate :check_reserved_at_is_future_or_present
   validate :check_appointment_availability, on: :create
   validate :check_duplicate_appointment, on: :create
+  before_update :check_past_appintment_when_canceled
```

Comment:
![ask](https://img.shields.io/badge/review-ask-blue)
validationではなく `before_update`にしたのってなんでですっけ？

Reply:
ありがとうございます！
ここではappoitnmentのstatusが更新されるときだけ確認すれば良いかなと思い、`before_update`にしました。

Reply:
処理の内容的にはvalidationなのかなという印象なので、以下にできませんか？

```suggestion
  validate :check_past_appintment_when_canceled on: :update
```

Reply:
validationにするとモデルでのテストがやりやすくなると思います（上記の変更をするならテストも追記してもらえると）

Reply:
ありがとうございます！
ご指摘の通り、
`validate :check_past_appintment_when_canceled on: :update`
に変更しておきます。
また過去の予約をcancelできないことを確認するテストも
spec/requests/appointments_spec.rbに記述しています。